StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/GenericAnalyzerTest.cs

```diff
@@ -0,0 +1,1191 @@
+    using Xunit;
+
+    internal abstract class GenericAnalyzerTest
+    {
+        private const int DefaultNumberOfIncrementalIterations = -1000;
+
+        private static readonly string DefaultFilePathPrefix = "Test";
+        private static readonly string CSharpDefaultFileExt = "cs";
+        private static readonly string VisualBasicDefaultExt = "vb";
+        private static readonly string CSharpDefaultFilePath = DefaultFilePathPrefix + 0 + "." + CSharpDefaultFileExt;
+        private static readonly string VisualBasicDefaultFilePath = DefaultFilePathPrefix + 0 + "." + VisualBasicDefaultExt;
+        private static readonly string TestProjectName = "TestProject";
+
+        private static readonly Lazy<IExportProviderFactory> ExportProviderFactory;
+
+        static GenericAnalyzerTest()
+        {
+            ExportProviderFactory = new Lazy<IExportProviderFactory>(
+                () =>
+                {
+                    var discovery = new AttributedPartDiscovery(Resolver.DefaultInstance, isNonPublicSupported: true);
+                    var parts = Task.Run(() => discovery.CreatePartsAsync(MefHostServices.DefaultAssemblies)).GetAwaiter().GetResult();
+                    var catalog = ComposableCatalog.Create(Resolver.DefaultInstance).AddParts(parts);
+
+                    var configuration = CompositionConfiguration.Create(catalog);
+                    var runtimeComposition = RuntimeComposition.CreateRuntimeComposition(configuration);
+                    return runtimeComposition.CreateExportProviderFactory();
+                },
+                LazyThreadSafetyMode.ExecutionAndPublication);
+        }
+
+        /// <summary>
+        /// Gets the language name used for the test.
+        /// </summary>
+        /// <value>
+        /// The language name used for the test.
+        /// </value>
+        public abstract string Language
+        {
+            get;
+        }
+
+        public string TestCode
+        {
+            set
+            {
+                if (this.TestSources.Count == 1)
+                {
+                    this.TestSources[0] = value;
+                }
+                else
+                {
+                    Assert.Empty(this.TestSources);
+                    this.TestSources.Add(value);
+                }
+            }
+        }
+
+        public List<string> TestSources { get; } = new List<string>();
+
+        public List<DiagnosticResult> ExpectedDiagnostics { get; } = new List<DiagnosticResult>();
+
+        public List<DiagnosticResult> RemainingDiagnostics { get; } = new List<DiagnosticResult>();
+
+        public List<DiagnosticResult> BatchRemainingDiagnostics { get; } = new List<DiagnosticResult>();
+
+        public List<string> DisabledDiagnostics { get; } = new List<string>();
+
+        public int? CodeFixIndex { get; set; }
+
+        public string FixedCode
+        {
+            set
+            {
+                if (this.FixedSources.Count == 1)
```

Comment:
💭 I don't think this should allow `FixedSources` to have a value. Otherwise this becomes possible with unexpected results:

```csharp
new CSharpTest
{
  TestSources = { firstSource },
  TestCode = secondSource,
}
```

Reply:
➡️ Moved to https://github.com/dotnet/roslyn-sdk/issues/202